pinctrl/hw_pwm.go

```diff
@@ -1,7 +1,7 @@
 //go:build linux
 
-// Package pi5 implements pi5
-package pi5
```

Comment:
this file still has alot of raspberry pi specific fields, and some comments calling out needing to generalize stuff. for now I was going to skip further generalizing this code

Reply:
is the end goal to fully generalize the pinctrl package?

Reply:
for this project I do not believe so, we just wanted to leave the majority of the pinctrl code outside of the raspberry pi repo so my goal is to refactor to the point where that is possible. 

but for a later project yes we should look to fully generalize pinctrl

Reply:
okay makes sense thanks. Is the pi5 code being moved to the raspberry-pi repo and then only the pinctrl package will live in this repo?

Reply:
yes. Ideally I only have to move the board.go, data.go, and config.go files to the other repo and this repo will only have the pinctrl package.

depending on how it ends up it might be worth leaving a pi5 implementation in this repo as a template for other boards, but going to save that until after I get the raspberry pi repo into a good state